src/emqx_hooks.erl

```diff
@@ -131,12 +135,24 @@ add(HookPoint, Action, Filter, Priority) when is_integer(Priority) ->
     add(HookPoint, #callback{action = Action, filter = Filter, priority = Priority}).
 
 %% @doc Like add/2, it register a callback, discard 'already_exists' error.
--spec(put(hookpoint(), action() | #callback{}) -> ok).
-put(HookPoint, Callback) ->
+-spec put(hookpoint(), action() | callback()) -> ok.
```

Comment:
Plz follow the previous code style

Reply:
Maybe the type def `callback/0` is better. (for language server to find the refs)